### PR TITLE
Add max number to profilers

### PR DESCRIPTION
**Pull request by chenhao94**

### CODE REVIEW

File: src/timer/timer_enable.cc [hunk 1]
```diff
@@ -186,15 +188,21 @@ void TimerStatCollector::Report(std::size_t index, cr_duration_nsec_t duration)
         return;
     }
 
-    auto& bucket = collector_entries_->at(index).duration_buckets_[bucket_idx];
+    auto& collector_entry = collector_entries_->at(index);
+    auto& bucket          = collector_entry.duration_buckets_[bucket_idx];
     ++bucket.hits_;
     bucket.total_duration_ns_ += duration;
+    collector_entry.max_duration_ns_ = std::max(collector_entry.max_duration_ns_, duration);
```
[the remark this answers is not in the document]
Comment (chenhao94) [reply src/timer/timer_enable.cc:195]: It is locked. You will see it if expand the lines above.

**Review by xkszltl — APPROVED** [review]

File: src/timer/timer_enable.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (xkszltl) [reply src/timer/timer_enable.cc:195]: Then it sounds slow to me....

[the remark this answers is not in the document]
Comment (xkszltl) [reply src/timer/timer_enable.cc:195]: Then it sounds slow to me....

[the remark this answers is not in the document]
Comment (chenhao94) [reply src/timer/timer_enable.cc:195]: It is `try_lock` so it won't block any execution. Suggestions are welcome though.

[the remark this answers is not in the document]
Comment (xkszltl) [reply src/timer/timer_enable.cc:195]: Close for now to merge, we can discuss later on.

**Pull request merged**